### DISCUSSION TITLE
TradeTracker not showing product ID in reference

### DISCUSSION THREAD

**Discussion by JulianQuispel** [datafeedr/wordpress-plugins#14] [Help] [answered]
Hi all,

in order for us to get more insight in which products sell well, we added the snippet of this page in the [docs](https://datafeedrapi.helpscoutdocs.com/article/212-tracking-ids). However, when clicking one of the links in the comparison set the reference doesn't appear in TradeTracker. I'm wondering whether this is a bug on our website or that tracking ID's are not included when clicking on one of the buttons in the comparison set.

Cheers,

Julian

**Comment by EricBusch** [comment, marked as answer]
Hi @JulianQuispel 

Can you share the exact code you are using? 

Also, can you provide a link to the page with your Comparison Sets?

Thanks
Eric

**Reply by JulianQuispel** [reply]
Hi @EricBusch

It's the code from the documentation.

```
/**
 * Add product ID to affiliate link for tracking.
 */
add_filter('dfrapi_tracking_id', function ($tracking_id, $product) {
  return wc_get_product_id_by_sku($product['_id']);
}, 20, 2);
```

An example of the comparison set can be found [here](https://gamespider.nl/product/activision-blizzard-call-of-duty-black-ops-cold-war-xbox-series-xbox-one-3/)

Thanks
Julian

**Reply by EricBusch** [reply]
Hi

The reason it's not adding any tracking ID is because [that product](https://gamespider.nl/product/activision-blizzard-call-of-duty-black-ops-cold-war-xbox-series-xbox-one-3/) no longer seems available in your store.

Hope this helps!

Eric

**Reply by JulianQuispel** [reply]
Hi @EricBusch 

Apologies. The slug seems to have changed. The correct URL is https://gamespider.nl/product/activision-blizzard-call-of-duty-black-ops-cold-war-xbox-series-xbox-one/

Thanks
Julian

**Reply by EricBusch** [reply]
What if you tried this code instead?

```php
add_filter('dfrapi_tracking_id', function ($tracking_id, $product) {
  return get_the_ID();
}, 20, 2);
```

**Reply by JulianQuispel** [reply]
@EricBusch thanks that seems to work!

**Reply by EricBusch** [reply]
You're welcome! Glad it worked!
